sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetFileFormat.scala

```diff
@@ -270,7 +270,7 @@ class ParquetFileFormat
       lazy val footerFileMetaData =
         ParquetFileReader.readFooter(sharedConf, filePath, SKIP_ROW_GROUPS).getFileMetaData
       // Try to push down filters when filter push-down is enabled.
-      val pushed = if (enableParquetFilterPushDown) {
+      val pushed = if (enableParquetFilterPushDown && filters.nonEmpty) {
```

Comment:
BTW, I don't think we save touching footer in this case because it will be read anyway later `footerFileMetaData`. But logically seems fine.

Reply:
U are right, at present, it is impossible to avoid this footer reading, footer cache may be more efficient, I will close this pr